qiita_db/metadata_template/prep_template.py

```diff
@@ -851,20 +851,30 @@ def _get_predecessors(workflow, node):
             starting_job = None
             pt_artifact = self.artifact.artifact_type
 
-        workflows = []
+        all_workflows = [wk for wk in qdb.software.DefaultWorkflow.iter()]
+        # are there any workflows with parameters?
+        check_requirements = False
+        default_parameters = {'prep': {}, 'sample': {}}
+        if [wk for wk in all_workflows if wk.parameters != default_parameters]:
+            check_requirements = True
         ST = qdb.metadata_template.sample_template.SampleTemplate
-        for wk in qdb.software.DefaultWorkflow.iter():
+        workflows = []
+        for wk in all_workflows:
             if wk.artifact_type == pt_artifact and pt_dt in wk.data_type:
+                if check_requirements and wk.parameters == default_parameters:
```

Comment:
Just to make sure, '==' has higher precedence than 'and', so the above two statements are:
if ((wk.artifact_type == pt_artifact) and (pt_dt)) in wk.data_type:
                 if ((check_requirements) and (wk.parameters == default_parameters)):

Is this how you want it?

Reply:
Yes, took me a minute to remember but yes - if we need to check for requirements and the `wk.parameters` are `==` to `default_parameters`, then continue. In other words, when `check_requirements` is False do not skip anything but if is True check the values in wk.parameters to see if we should skip things below. 

Anyway, you comment actually made me realize that we don't need the lower code `check_requirements = True`; I'll remove them.